src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -25,8 +25,15 @@ internal RuntimeTypeHandle GetNativeHandle() =>
         internal RuntimeType GetTypeChecked() =>
             m_type ?? throw new ArgumentNullException(null, SR.Arg_InvalidHandle);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern bool IsInstanceOfType(RuntimeType type, [NotNullWhen(true)] object? o);
+        internal static bool IsInstanceOfType(RuntimeType type, [NotNullWhen(true)] object? o)
```

Comment:
Is this only used from `RuntimeType.IsInstanceOfType` ?

We should delete this method and make `RuntimeType.IsInstanceOfType` to call `CastHelpers.IsInstanceOfAny` directly on CoreCLR.

Reply:
> Is this only used from `RuntimeType.IsInstanceOfType` ?

Yes.

> We should delete this method and make `RuntimeType.IsInstanceOfType` to call `CastHelpers.IsInstanceOfAny` directly on CoreCLR.

We would need to do this will Mono and native AOT too, I think. The definition of `RuntimeType.IsInstanceOfType` is under SPCL source under `libraries/`. Am I missing something?

Reply:
Some methods on RuntimeType.cs under libraries just forward to static method on RuntimeTypeHandle that does not make sense. It would make more sense to just implement the method directly on Runtime.

NAOT does not use RuntimeType.cs implementation under libraries, so it is just about Mono. To avoid actually changing Mono in this PR, you can move the no longer shared implementation of `RuntimeType.IsInstanceOfType` from under libraries to RuntimeType.Mono.cs.

Reply:
Perfect, that is what I was confirming.